app/scripts/views/settings/recovery_codes.js

```diff
@@ -0,0 +1,142 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const $ = require('jquery');
+const Cocktail = require('cocktail');
+const FormView = require('../form');
+const ModalSettingsPanelMixin = require('../mixins/modal-settings-panel-mixin');
+const Template = require('templates/settings/recovery_codes.mustache');
+const RecoveryCodePrintTemplate = require('templates/settings/recovery_codes_print.mustache');
+const RecoveryCode = require('../../models/recovery-code');
+
+var t = FormView.t;
+
+const View = FormView.extend({
+  template: Template,
+  className: 'recovery-codes',
+  viewName: 'settings.two-step-authentication.recovery-codes',
+
+  events: {
+    'click .copy-codes': FormView.preventDefaultThen('_copyCodes'),
+    'click .download-codes': '_downloadCodes',
+    'click .print-codes': FormView.preventDefaultThen('_printCodes'),
+    'click .replace-codes-link': FormView.preventDefaultThen('_replaceRecoveryCodes'),
+    'click .two-step-authentication-done': FormView.preventDefaultThen('_returnToTwoStepAuthentication')
+  },
+
+  _returnToTwoStepAuthentication() {
+    this.navigate('settings/two_step_authentication');
+  },
+
+  _copyCodes() {
+    // This copies the recovery codes to clipboard by creating a tiny transparent
+    // textArea with recovery code contents. Then it executes the
+    // browser `copy` command.
+    $('<textArea class=\"recovery-code-text-area\"></textArea>').appendTo('#recovery-codes');
+    this.$('.recovery-code-text-area').html(this.recoveryCodesText);
+    this.$('.recovery-code-text-area').select();
+    this.$('.recovery-code-text-area').focus();
+    try {
+      this.window.document.execCommand('copy');
+      this._displaySuccess(t('Codes copied'));
+    } catch (err) {
+      this._displayError(t('Failed to copy codes'));
```

Comment:
Can the user remedy this situation somehow? Maybe suggest they select the text and use the browser's copy to clipboard functionality?